metabase/metabase
# New Action Dashcard Modal + Sidebar

resolves #27916

## Description

- Updates the Action Sidebar to edit the properties of a single action dashcard
- Removes the ability to launch the visualization settings modal for action dashcards (because it would duplicate what's in the sidebar)
- Adds a new `ActionDashcardSettingsModal` where you can select/change the action assigned to a button, and map parameters.
- significantly reworks the parameter mapping components, because they live in a very different place now
- doesn't add any e2e tests for now, since we'll be able to do a much better job of it once https://github.com/metabase/metabase/pull/27989 merges

. | . 
--- | ---
add | ![addAction0203](https://user-images.githubusercontent.com/30528226/216677238-608475e6-276f-4538-bf93-2c87c90cbbd5.gif)
update | ![updateAction0203](https://user-images.githubusercontent.com/30528226/216677267-454f52aa-9c07-48c8-b612-e067b2ea197f.gif)


## How to Test

- make sure you have at least one model with actions (which requires a writable (non h2) db
- open a dashboard, enter edit mode, and click the add action button
- a new button should immediately be added to your dashboard and the action sidebar should open
- you should be able to link an action and parameters (if any) via the "pick an action" button
- once an action is linked, and you close the modal, you should be able use the sidebar to update your button text and variant
- you should be able to update the action settings for a button via the lightning bolt icon in the actions menu
